src/som/interpreter/actors/ResolvePromiseNode.java

```diff
@@ -37,23 +37,23 @@ public SResolver selfResolution(final SResolver resolver, final SPromise result,
   /**
    * Handle the case that a promise is resolved with another promise, which is not itself.
    */
-  @Specialization(guards = {"resolver.getPromise() != result"})
-  public SResolver chainedPromise(final VirtualFrame frame,
-      final SResolver resolver, final SPromise result,
+  @Specialization(guards = {"resolver.getPromise() != resolvedValue"})
+  public SResolver chainedPromise(final VirtualFrame frame, final SResolver resolver, final SPromise resolvedValue,
       final boolean isBreakpointOnPromiseResolution) {
     assert resolver.assertNotCompleted();
-    SPromise promise = resolver.getPromise();
+    SPromise promiseToBeResolved = resolver.getPromise();
 
-    synchronized (result) {
-      if (result.isResolvedUnsync()) {
-        normalResolution(frame, resolver, result.getValueUnsync(),
-            isBreakpointOnPromiseResolution);
-      } else if (result.isErroredUnsync()) {
+    synchronized (resolvedValue) {
+      if (resolvedValue.isResolvedUnsync()) { //resolved
+        normalResolution(frame, resolver, resolvedValue.getValueUnsync(),
+           isBreakpointOnPromiseResolution);
+      } else if (resolvedValue.isErroredUnsync()) {
         CompilerDirectives.transferToInterpreter();
         throw new NotYetImplementedException();
-      } else {
-        synchronized (promise) { // TODO: is this really deadlock free?
-          result.addChainedPromise(promise);
+      } else { //unresolved promise
```

Comment:
the test seems still pretty clear to me. I would avoid such trivial comments if possible.